src/main/java/frc/trigon/robot/misc/objectdetectioncamera/SimulationObjectDetectionCameraIO.java

```diff
@@ -40,127 +30,224 @@ protected SimulationObjectDetectionCameraIO(String hostname) {
-        for (Translation2d objectPlacement : objectsOnField) {
-            if (objectPlacement.getDistance(robotTranslation) <= PICKING_UP_TOLERANCE_METERS) {
-                heldObject = new Pose3d[]{getHeldObjectPose(robotPose)};
-                objectsOnField.remove(objectPlacement);
-                Commands.getDelayedCommand(10, () -> objectsOnField.add(objectPlacement)).schedule();
-                break;
+        List<Translation2d> visibleObjectsPlacements = getVisibleObjectsPlacements(robotPose);
+
+        for (Translation2d objectPlacement : visibleObjectsPlacements) {
+            final double robotDistanceToObject = getObjectDistance(objectPlacement, robotPose);
+            if (robotDistanceToObject < closestObjectDistance) {
+                closestObject = objectPlacement;
+                closestObjectYaw = getRobotYawToObject(objectPlacement, robotPose).minus(robotPose.getRotation());
+                closestObjectDistance = robotDistanceToObject;
             }
         }
-    }
 
-    private Pose3d getHeldObjectPose(Pose2d robotPose) {
-        //todo: get the pose of the object held by the robot
-        return new Pose3d();
+        logObjectPlacement(closestObject);
+        return closestObjectYaw;
     }
 
-    private Transform3d toTransform(Pose3d pose) {
-        return new Transform3d(pose.getTranslation(), pose.getRotation());
-    }
+    /**
+     * Gets the placements of all visible objects by checking if they are within range and within the horizontal FOV.
+     *
+     * @param robotPose the position of the robot on the field
+     * @return the placements of the visible objects
+     */
+    private List<Translation2d> getVisibleObjectsPlacements(Pose2d robotPose) {
+        final List<Translation2d> visibleObjects = new ArrayList<>();
+        int currentIndex = 0;
 
-    private Pose3d[] toPosesArray(List<Translation2d> translationsList) {
-        final Pose3d[] posesArray = new Pose3d[translationsList.size()];
-        for (int i = 0; i < translationsList.size(); i++) {
-            final Translation2d translation = translationsList.get(i);
-            posesArray[i] = new Pose3d(translation.getX(), translation.getY(), 0.1, new Rotation3d());
+        for (Translation2d currentObject : objectsOnField) {
+            final Rotation2d robotYawToObject = getRobotYawToObject(currentObject, robotPose);
+            if (!isWithinHorizontalFOV(robotYawToObject, robotPose) || !isWithinDistance(objectsOnField.get(0), robotPose))
+                continue;
+            visibleObjects.add(currentIndex, currentObject);
+            currentIndex++;
         }
-        return posesArray;
+        return visibleObjects;
+    }
+
+    /**
+     * Gets the yaw of an object relative to the robot.
+     *
+     * @param objectPlacement the placement of the object on the field
+     * @param robotPose       the position of the robot on the field
+     * @return the yaw of the object relative to the robot
```

Comment:
not robot, camera
add some kind of way to handle the fact the camera isn't always in the front of the robot